CDS-57073 Service Dashboard revamp - Phase 2 (#1984)
harness/developer-hub: docs/continuous-delivery/monitor-deployments/monitor-cd-deployments.md

## Services dashboard

**What is a Harness service?** A Harness service is both a logical and configuration view of the services you deploy.

The logical view is immutable and contains a name, unique Id, and description. The configuration view can be changed with each stage of a pipeline and contains the artifacts, manifests, repo URLs, etc., for that stage's deployment of the service.

**What is a service instance in Harness?** Harness licensing is determined by the service instances you deploy. A service instance is when you use Harness to deploy the service and the underlying infrastructure for the instance. For example, an instance of a Kubernetes workload where Harness creates the pods.

The **Services** dashboard provides an overview of all the services and service instances in your project:

<docimage path={require('./static/monitor-cd-deployments-19.png')} width="60%" height="60%" title="Click to view full size image" />

## Individual service dashboards

:::note

Currently, the following service dashboard functionality is behind the feature flag `CDC_SERVICE_DASHBOARD_REVAMP_NG`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.

:::

> Click a service in the **Total Services** table drills down to show more service details:

Reviewer: ```suggestion
Click a service in the **Total Services** table to drill down and show more service details:
```

<docimage path={require('./static/eebdfe6a596f69e0fd2dca49ce04dd5659094aab8d76ef38886fa905659ffac9.png')} width="60%" height="60%" title="Click to view full size image" />  

An individual service's dashboard includes the following features.

### Open tasks

The **See Open Tasks** banner will appear when there are deployments with the following statuses:

- Aborted
- Aborted by deployment freeze
- Expired
- Waiting
- Failed

Selecting **See Open Tasks** displays executions that require your attention.

> The **See Open Tasks** button is only displayed if there are tasks over the last 3 days.

Reviewer: You're calling it a button here but a banner before the list...are these two different UI elements?

If the **See Open Tasks** button is not displayed, you can see all tasks using the bell icon also.

<docimage path={require('./static/9f67ae3af9c7f548f8bd9f802008e0d5d8710b167c9c4a00d4e8b611aa370401.png')} width="60%" height="60%" title="Click to view full size image" />

### Environments tab

The **Environments** tab lists the environments and environment groups where the service has been deployed by one or more pipelines.

<docimage path={require('./static/4e748162f05ea1d64ee5c64ca924fc41afe6c61fce81ccbbd7f1d32a64b40a90.png')} width="60%" height="60%" title="Click to view full size image" />  

Each environment card shows the following:

- **Latest artifact:** the latest artifact deployed by this service to this environment.
- **Drift detection:** drift detection displays any differences between the deployments to environments in an environment group. For example, in this drift detection, two environments are using different artifacts with this service:

  <docimage path={require('./static/2f326c7718023be7ad2226f2ea3adb8a8eb772b7407304d134bbb3ffc8e473ad.png')} width="60%" height="60%" title="Click to view full size image" />  

>   Drift detection is helpful to detect inconsistencies. For example, when you deploy a service to an environment group and there are failures in one of the environments or someone performs a different deployment to one or more of the group's environments.

Reviewer: ```suggestion
  Drift detection is helpful to detect inconsistencies. For example, you deploy a service to an environment group and there are failures in one of the environments. Another example: someone performs a different deployment to one or more of the group's environments.
```

  - Clicking the artifact name opens details for all environments where the artifact was deployed. 
  - Clicking the environment opens details for the specific environment where the artifact was deployed.
- **Drift detection:** drift detection displays any differences between the deployments to environments in an environment group. For example, if the artifact you select was deployed to an environment group, but one of the group's environments deployed a different artifact, it is highlighted as drift:

  <docimage path={require('./static/b46ca573dd4b036c3ab5189e6edda2fd86207f6401e8337a52f365da443007a9.png')} width="60%" height="60%" title="Click to view full size image" />  

  Clicking on the environment group name will take you to that group's settings. Clicking on the execution date (in this example, `1 day ago`) will take you to the pipeline execution details.
- **Deployments:** each deployment of the service or a selected artifact. If you filter using **My Deployments** you see only those deployments that your user account initiated. 

### View in table

The **View in Table** setting shows all of the instances of the service in all of the environments used with the service. 

> 

Reviewer 2: we can also add here as note that we show latest information of environment in table view
For example if after instance deployment a environment name got changed, then we will show the latest name, similiarly for other fields

![](../cd-dashboards/static/monitor-cd-deployments-24.png)

> Clicking on data points or service and project names brings up data tables.

Reviewer: ```suggestion
Clicking on data points, service names,  or project names brings up data tables.
```
